### PR TITLE
Neat 494 extend .show for information rules

### PR DESCRIPTION
**Pull request by nikokaoja**
### Added
- Support for information rules and instance plotting in NeatSession
Screenshot 2024-10-28 at 16 16 40

### Improved
- Better plotting of rules for dms and information rules in NeatSession (accounts for `subClassOf` and `implements`)

Screenshot 2024-10-28 at 16 17 34

### CODE REVIEW

File: cognite/neat/_session/_show.py [hunk 1]
```diff
@@ -14,26 +18,147 @@ class ShowAPI:
...
+        """
+
+        G = nx.DiGraph()
+
+        types = set()
+
+        for (
+            s,
+            p,
+            o,
```
Comment (doctrino) [inline cognite/neat/_session/_show.py:66]: subject, predicate, object?

File: cognite/neat/_session/_show.py [hunk 2]
```diff
@@ -14,26 +18,147 @@ class ShowAPI:
...
+
+    def __call__(self) -> Any:
+        if not self._state.store.graph:
+            print("No graph data available.")
+            return None
+
+        widget = CytoscapeWidget()
+        widget.layout.height = "700px"
+
+        G, types = self._get_instance_di_graph_and_types()
```
Comment (doctrino) [inline cognite/neat/_session/_show.py:36]: G = networkx Graph?

File: cognite/neat/_session/_show.py [hunk 3]
```diff
@@ -14,26 +18,147 @@ class ShowAPI:
...
+
+
+class ShowInstanceAPI:
+    def __init__(self, state: SessionState) -> None:
+        self._state = state
+
+    def __call__(self) -> Any:
+        if not self._state.store.graph:
+            print("No graph data available.")
+            return None
```
Comment (doctrino) [inline cognite/neat/_session/_show.py:31]: Suggest just assuming the graph is available and raise an error with hint on how to load data if it is not there.

File: cognite/neat/_session/_show.py [hunk 4]
```diff
@@ -14,26 +18,147 @@ class ShowAPI:
...
+        G, types = self._get_instance_di_graph_and_types()
+        widget_style = self._generate_cytoscape_widget_style(types)
+        widget.set_style(widget_style)
+
+        widget.graph.add_graph_from_networkx(G)
+        print("Max of 100 nodes and edges are displayed, which are randomly selected.")
+
+        return display(widget)
+
+    def _get_instance_di_graph_and_types(self):
```
Comment (doctrino) [inline cognite/neat/_session/_show.py:45]: Should this be somewhere else? Or is it specific to the visualization. Seems generic to create a networkx graph from the triple store

File: cognite/neat/_session/_show.py (hunk 1 above)
Comment (doctrino) [reply cognite/neat/_session/_show.py:66]: This is probably just my skepticism to single letter names, the code is quite readable so up for debate whether to change it.

File: cognite/neat/_session/_show.py [hunk 5]
```diff
@@ -14,26 +18,147 @@ class ShowAPI:
...
 
 
 class ShowDataModelAPI:
     def __init__(self, state: SessionState) -> None:
         self._state = state
 
     def __call__(self) -> Any:
+        if not self._state.last_verified_dms_rules and not self._state.last_verified_information_rules:
+            print("No rules have been verified yet.")
+            return None
```
Comment (doctrino) [inline cognite/neat/_session/_show.py:143]: Lets discuss this one, I think there is a pattern were we just raise NeatSession errors that are picked up and displayed nicely.

**Review by doctrino — APPROVED** [review]
Great to have nice visualizations. I think we should have a discussion on error patterns, as well as were to keep the analysis/move into networkx in the code.

File: cognite/neat/_session/_show.py (hunk 1 above)
Comment (nikokaoja) [reply cognite/neat/_session/_show.py:66]: good point, was quick , I put right names now

File: cognite/neat/_session/_show.py (hunk 2 above)
Comment (nikokaoja) [reply cognite/neat/_session/_show.py:36]: renamed it to NxGraph so we know it is Networkx Graph

File: cognite/neat/_session/_show.py (hunk 4 above)
Comment (nikokaoja) [reply cognite/neat/_session/_show.py:45]: we will move it to `probe`

**Pull request merged**